File: 924/main.go
package main

import "fmt"

func main() {
	list := make([]int, 0)
	list = append(list, 1)
	fmt.Println(list)
	fmt.Println(minMalwareSpread([][]int{
		{1, 1, 0},
		{1, 1, 0},
		{0, 0, 1},
	}, []int{0, 1}))
}

func minMalwareSpread(graph [][]int, initial []int) int {
	n := len(graph)
	ids := make(map[int]int)
	idToSize := make(map[int]int)
	id := 0
	for i := range ids {
		if ids[i] == 0 {
			id++
			ids[i] = id
			size := 1
			q := []int{i}
			for len(q) > 0 {
				u := q[0]
				q = q[1:]
				for v := range graph[u] {
					if ids[v] == 0 && graph[u][v] == 1 {
						size++
						q = append(q, v)
						ids[v] = id
					}
				}
			}
			idToSize[id] = size
		}
	}
	idToInit := make(map[int]int)
	for _, u := range initial {
		idToInit[ids[u]]++
	}
	ans := n + 1
	ansRemoved := 0
	for _, u := range initial {
		removed := 0
		if idToInit[ids[u]] == 1 {
			removed = idToSize[ids[u]]
		}
		if removed > ansRemoved || (removed == ansRemoved && u < ans) {
			ans, ansRemoved = u, removed
		}
	}
	return ans
}

File: 924/main.rs
use std::collections::{HashMap, VecDeque};

#[derive(Debug)]
struct Solution {}

impl Solution {
    pub fn min_malware_spread(graph: Vec<Vec<i32>>, initial: Vec<i32>) -> i32 {
        let n = graph.len();
        let mut ids: Vec<i32> = vec![0; n];
        let mut id_to_size: HashMap<i32, i32> = HashMap::new();
        let mut id = 0;
        for i in 0..n {
            if ids[i] == 0 {
                id += 1;
                let mut q = VecDeque::from([i]);
                ids[i] = id;
                let mut size = 1;
                while let Some(u) = q.pop_front() {
                    for v in 0..n {
                        if graph[u][v] == 1 && ids[v] == 0 {
                            size += 1;
                            q.push_back(v);
                            ids[v] = id;
                        }
                    }
                }
                id_to_size.insert(id, size);
            }
        }
        let mut id_to_initials: HashMap<i32, i32> = HashMap::new();
        for &u in initial.iter() {
            *id_to_initials.entry(ids[u as usize]).or_insert(0) += 1;
        }
        let mut ans = n as i32 + 1;
        let mut ans_removed = 0;
        for &u in initial.iter() {
            let removed = if id_to_initials[&ids[u as usize]] == 1 {
                id_to_size[&ids[u as usize]]
            } else {
                0
            };
            if removed > ans_removed || (removed == ans_removed && u < ans) {
                ans = u;
                ans_removed = removed;
            }
        }
        ans
    }
}

fn main() {
    println!("main");
    let initial = vec![1, 2];
    let g = vec![vec![1, 1, 1], vec![1, 1, 1], vec![1, 1, 1]];
    println!("{}", Solution::min_malware_spread(g, initial));
}
